airbyte-integrations/connectors/source-us-census/source_us_census/components.py

```diff
@@ -132,4 +135,46 @@ def interpret_response(self, response_or_exception: Optional[Union[requests.Resp
         default_reponse_filter = DefaultHttpResponseFilter(parameters={}, config=self.config)
         default_response_filter_resolution = default_reponse_filter.matches(response_or_exception)
 
-        return default_response_filter_resolution if default_response_filter_resolution else DEFAULT_ERROR_RESOLUTION
+        return (
+            default_response_filter_resolution
+            if default_response_filter_resolution
+            else create_fallback_error_resolution(response_or_exception)
+        )
+
+
+@dataclass
+class USCensusSchema(SchemaLoader):
+    """
+    The US Census website hosts many APIs: https://www.census.gov/data/developers/data-sets.html
+
+    These APIs return data in a non standard format.
+    We create the JSON schemas dynamically by reading the first "row" of data we get.
+
+    In this implementation all records are of type "string", but this function could
+    be changed to try and infer the data type based on the values it finds.
+    """
+
+    config: Config
+
+    def get_json_schema(self) -> Mapping[str, Any]:
```

Comment:
@topefolorunso can you add some unit tests to verify this works as expected?

Reply:
Tests added now @girarda